src/vm/database/clarity_db.rs

```diff
@@ -143,84 +202,42 @@ impl <'a> ClarityDatabase <'a> {
     }
 }
 
-// Simulating blocks
+// Get block information
 
 impl <'a> ClarityDatabase <'a> {
-    fn get_simmed_block(&mut self, block_height: u64) -> SimmedBlock {
-        let key = ClarityDatabase::make_key_for_trip(
-            &QualifiedContractIdentifier::transient(), StoreType::SimmedBlock, &block_height.to_string());
-        self.get(&key)
-            .expect("Failed to obtain the block for the given block height.")
-    }
-    
-    pub fn get_simmed_block_height(&mut self) -> u64 {
-        let key = ClarityDatabase::make_key_for_trip(
-            &QualifiedContractIdentifier::transient(), StoreType::SimmedBlockHeight, ":dummy:");
-        self.get(&key)
-            .expect("Failed to obtain block height.")
+    pub fn get_index_block_header_hash(&mut self, block_height: u32) -> BlockHeaderHash {
+        self.store.get_block_header_hash(block_height)
+        // the caller is responsible for ensuring that the block_height given
+        //  is < current_block_height, so this should _always_ return a value.
+            .expect("Block header hash must return for provided block height")
     }
 
-    pub fn get_simmed_block_time(&mut self, block_height: u64) -> u64 {
-        self.get_simmed_block(block_height)
-            .block_time
-    }
-    pub fn get_simmed_block_header_hash(&mut self, block_height: u64) -> BlockHeaderHash {
-        BlockHeaderHash::from_bytes(&self.get_simmed_block(block_height)
-                                    .block_header_hash).unwrap()
-    }
-    pub fn get_simmed_burnchain_block_header_hash(&mut self, block_height: u64) -> BurnchainHeaderHash {
-        BurnchainHeaderHash::from_bytes(&self.get_simmed_block(block_height)
-                                        .burn_chain_header_hash).unwrap()
-    }
-    pub fn get_simmed_block_vrf_seed(&mut self, block_height: u64) -> VRFSeed {
-        VRFSeed::from_bytes(&self.get_simmed_block(block_height).vrf_seed).unwrap()
+    pub fn get_current_block_height(&mut self) -> u32 {
+        self.store.get_current_block_height()
     }
 
-    fn set_simmed_block_height(&mut self, block_height: u64) {
-        let block_height_key = ClarityDatabase::make_key_for_trip(
-            &QualifiedContractIdentifier::transient(), StoreType::SimmedBlockHeight, ":dummy:");
-        self.put(&block_height_key, &block_height);
+    pub fn get_block_header_hash(&mut self, block_height: u32) -> BlockHeaderHash {
```

Comment:
What happens when a contract tries to get the block header hash for a block height that's too high?

Reply:
Same question for all these un-simmed methods.

Reply:
If it gets to this function, it panics. The caller of these functions performs a check on the block_height and returns None if the block_height is >= the current block height.

Reply:
Alright, just making sure we're not setting ourselves up for unexpected runtime panics :+1: